src/libs/actions/Session.js

```diff
@@ -201,7 +201,7 @@ function resendValidationLink() {
  */
 function resetPassword() {
     Onyx.merge(ONYXKEYS.ACCOUNT, {loading: true, forgotPassword: true});
-    API.ResetPassword({email: credentials.login})
+    API.ResendValidateCode({email: credentials.login})
```

Comment:
Hmm, i don't quite follow this change. We use ResendValidateCode when an account is not validated right and hence we want to resend them the validate code. But instead in this flow the account is validated and has just forgotten their password so using the endpoint ResendValidateCode incorrectly only to let the API continue the code with a validated account [here](https://github.com/Expensify/Auth/pull/5468/files#diff-7b6cd1030cc3fe6a8bbc09ce2c558b5e91f8df9099f648aae3f59c7f9900f5e8R47) is really weird. 

I'm not a fan of changing the fundamental behavior of an endpoint just because the request is coming from ecash. How about this. We don't make any change to ecash and instead in auth over [here](https://github.com/Expensify/Auth/blob/9cb4470dddf2c908ea41a34929b8c4ecddb4a734/auth/command/ResetValidateCode.cpp#L58) such that if the request is from ecash we use the new setCashPassword template that you created. Just have to pass accountID as well to the template as well and then it should just work perfectly fine.

At the same time then i think this [code block](https://github.com/Expensify/Web-Expensify/blob/86c3d3c7cee4b3abd44b9e851b56f0e6ad8c9432/lib/Notification/ResetValidateCode.php#L21-L22) can be updated to remove the `isFromExpensifyCash` bits since we then won't be hitting that notification with `isFromExpensifyCash`. 

Thoughts?

Reply:
<details>
<summary>Tracing what dot com does</summary>

```
- Looking at what dot com does we call the API with ResetPassword 
    - https://github.com/Expensify/Web-Expensify/blob/f006df8cb25d14fa5ce2ce72cc031c2e969ce854/homepage/js/signin.jsx#L530 
    - https://github.com/Expensify/Web-Expensify/blob/f006df8cb25d14fa5ce2ce72cc031c2e969ce854/api.php#L1171
- which triggers UserAPI::resetPassword https://github.com/Expensify/Web-Expensify/blob/f006df8cb25d14fa5ce2ce72cc031c2e969ce854/lib/UserAPI.php#L2404
 ```
</details>

I agree with Chirag we should keep the API call the same and hit `ResetPassword` not `ResendValidateCode`, I don't think any dot cash changes are needed.

And I agree with his proposal: 
- Update Auth to send the `setCashPassword` template when its from dotcash. 
- Clean up lib/Notification/ResetPassword.php in web-e.

Reply:
Yup, I totally agree! Felt weird adding an if check so I'm glad you picked up on this.

Reply:
@chiragsalian @bondydaa I'm going to close this particular PR since no changes are needed; I've requested your reviews on these PRs which should implement the changes you've described:
- https://github.com/Expensify/Auth/pull/5468
- https://github.com/Expensify/Web-Expensify/pull/30688